src/co_tools/co_fastq.py

```diff
@@ -0,0 +1,156 @@
+import os
+import re
+import subprocess
+import sys
+from glob import glob
+from pathlib import Path
+
+# local imports
+from .get_logger import LOGGER
+
+
+def get_fastq_pair(dir_path: str = "../data"):
+    total_dirs = 0
+    for base, dirs, files in os.walk(dir_path):
+        for dir in dirs:
+            total_dirs += 1
+    if total_dirs != 2:
+        LOGGER.error(
+            f"The fastq files in {dir_path} are not properly configured"
+            + " to use this function. There should be only 2 folders"
+            + " inside the data folder."
+        )
+        return 0
+    prefix_dict, this_prefix = {}, None
+    fwd, rev = None, None
+    for path in glob(str(f"{dir_path}/**/*.fastq.gz"), recursive=True):
+        if prefix := get_prefix(path):
+            if prefix in prefix_dict:
+                prefix_dict[prefix].append(path)
+                if len(prefix_dict[prefix]) == 3:
+                    LOGGER.info(
+                        f"prefix {prefix} occurs 3 times in the {dir_path} folder"
+                    )
+                    this_prefix = prefix
+                    break
+            else:
+                prefix_dict[prefix] = [path]
+        else:
+            LOGGER.warning(f"No prefix determined for {path}")
+    if not prefix_dict:
+        LOGGER.error(f"No files found in {dir_path}")
+        return 0
+    if not this_prefix:
+        LOGGER.error(f"fastq files in {dir_path} not properly organized")
+        return 0
+    for path in prefix_dict[this_prefix]:
+        if get_read_direction(path) == "1":
+            fwd = path
+        elif get_read_direction(path) == "2":
+            rev = path
+    if fwd and rev:
+        LOGGER.info(f"returning {fwd},{rev}")
+        return f"{fwd},{rev}"
+    else:
+        LOGGER.error(
+            f"Could not find complementary pair of fastq files in {dir_path}"
+        )
+        return 0
+
+
+def get_fwd_fastqs(dir: str = "../data"):
+    if some_fastq := (
+        subprocess.check_output(["find", "-L", dir, "-name", "*.fastq.gz"])
```

Comment:
This should not be a subprocess call. Use "glob" instead. 

https://docs.python.org/3/library/glob.html

Reply:
What benefit does glob bring in this particular instance? Performance?

Reply:
Yes and interoperability between systems, though we're pretty married to Linux. The other thing (and this is less a practical concern, but matters since this is public) is just that it looks a bit like code written by someone less familiar with Python since glob is a pretty core library.